kubectl/role.go

```diff
@@ -0,0 +1,81 @@
+package kubectl
+
+import (
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PrepareTillerRole will construct a new Role struct with the provided metadata. The role can later
+// be used to add rules.
+func PrepareRole(
+	namespace string,
+	name string,
+	labels map[string]string,
+	annotations map[string]string,
+	rules []rbacv1.PolicyRule,
+) *rbacv1.Role {
+	newRole := rbacv1.Role{}
+	newRole.Name = name
+	newRole.Namespace = namespace
+	newRole.Labels = labels
+	newRole.Annotations = annotations
+	newRole.Rules = rules
```

Comment:
Is there a reason you are doing it this way and not directly in the struct initialization?

Reply:
Can't use a struct literal here because several of the fields are promoted from [ObjectMeta](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta) and Go can't mix `field:value` and `value` initializers in a struct declaration.

Reply:
Ah gotcha forgot about that. Thanks for the clarification!